paper_sections/05-computational_and_statistical_issues.Rmd

```diff
@@ -628,9 +636,22 @@ to balance penalties between nested smooth terms (this is why nested random
 effects can be fitted). We have observed that **mgcv** still occasionally
 finds solutions with simulated data where the global term is over-smoothed.
 
-To avoid this issue, we recommend both careful choice of basis and setting model degrees of freedom so that groupwise terms are either slightly less flexible than the global term or have a smaller null space. In the examples in section III, we used smoothers with an unpenalized null space (standard thin plate splines) for the global smoother and ones with no null space for the groupwise terms[^gsnull]. When using thin plate splines, it may also help to use splines with a lower order of derivative penalized in the groupwise smoothers than the global smoothers, as lower-order "tp" splines have fewer basis functions in the null space. For example, we used `m=2` (penalizing squared second derivatives) for the global smoother, and `m=1` (penalizing squared first derivatives) for groupwise smoothers in models *GS* and *GI*. Another option is to use a lower number of basis functions (`k`) for groupwise relative to global terms.  This will reduce the maximum flexibility possible in the groupwise terms. We do caution that these are just rules of thumb. In cases where an accurately estimated global smoother is essential, we recommend either fitting model *G* or using specialized functional regression software such as the `fosr` function in the *refund* package [@scheipl_functional_2014], which allows the user to enforce constraints on the groupwise smoothers so that they always sum to zero at any given point (avoiding the collinearity issue). Also, see below for more information on functional regression.
+To avoid this issue, we recommend both careful choice of basis and setting model degrees of freedom so that group-level terms are either slightly less flexible than the global term or have a smaller null space. In the examples in section III, we used smoothers with an unpenalized null space for the global smoother and ones with no null space for the group-level terms. When using thin plate splines, this can be done with splines with a lower order of derivative penalized in the group-level smoothers than the global smoothers, as lower-order TPRS splines have fewer basis functions in the null space. For example, we used `m=2` (penalizing squared second derivatives) for the global smoother, and `m=1` (penalizing squared first derivatives) for group-level smoothers in models *GS* and *GI*. Another option is to use a lower number of basis functions (`k`) for group-level relative to global terms.  This will reduce the maximum flexibility possible in the group-level terms. We do caution that these are just rules of thumb. 
```

Comment:
I agree; it's not the best bet. What about if we take out all discussion of whether the global smoother should be more or less smooth than the group level smoothers? I don't think we know enough right now about how they interact to determine if one approach is universally better than the other, and I definitely don't think we're in a position to give accurate advice about how to choose relative dimension sizes.

Reply:
I thought the argument being made in the discussion was that if we make no statements about the properties of the global versus group-level smooths, then we may not be able to uniquely identify all the smooths. However, if we can imbue the model with some guidance as to the respective properties of the different smooths, then we might be able to identify them.

In other words, we need to include prior information as to the expected wiggliness, or differentiate due to the order of the penalty. I see this as user choice and not something we should be providing advice on, other than to point out to the user that they may need to think about the properties of the smooths they want to estimate.

Reply:
Good point as well. I've adjusted the text to try and spell out that this is a modelling choice. What do you think of the language I just pushed? 

> To avoid this issue, we recommend both careful choice of basis and relative degrees of freedom of global and group-level terms. In the examples in section III, we used smoothers with an unpenalized null space for the global smoother and ones with no null space for the group-level terms. When using thin plate splines, this can be done with splines with a lower order of derivative penalized in the group-level smoothers than the global smoothers, as lower-order TPRS splines have fewer basis functions in the null space. For example, we used `m=2` (penalizing squared second derivatives) for the global smoother, and `m=1` (penalizing squared first derivatives) for group-level smoothers in models *GS* and *GI*. Another option is to specify the maximum degrees of freedom (`k`) for the group-level smoother either substantially higher or lower than the global smoother; 
this is in effect an approximate way to specify a prior belief in the relative smoothness of the global versus group-level functions. 
If the group-level term is set to have greater `k` compared to the gobal term, this encodes the assumption that the global function should not be very wiggly, but group-level deviations from that smooth might vary from that, and vice versa if `k` is set lower for group-level terms than for the global smoother.